### ISSUE TITLE
Modify JS validations so that optional questions can be partially filled out and still submitted

### ISSUE THREAD

**Issue by lindsayzad** [seattle-uat/civiform#1561]
Related to the request to add optional questions: issue #1307 That code is in PR #1550

The existing JS validations are already compatible with a question not being filled out and being optional ( “Issue #1418 was closed by #1528, which makes the validation JS compatible with optional questions.” ), however, once an optional question begins to be answered it cannot be left partially filled.  

For example, only filling in the first name of a name question, or a partial address.  We should allow optional questions to be only partially filled and still submitted.  Also include a browser test where an optional question is partially filled

**Comment by melanie-exygy** [comment]
@kpeterson-exygy, I'm doing some issue triage and came across this request for a UX change.  What's your opinion on validation for partially completed optional questions?  My opinion, without having done any research, is that validation is still needed for optional questions because we wouldn't want to save incomplete user info.  For example, on an optional email question, someone shouldn't be able to submit the answer "melanie@" because that isn't a valid email.  
I'd like to close this issue, but wanted to check with you first about whether it seems like something that would be a UX enhancement.

**Comment by kpeterson-exygy** [comment]
@melanie-exygy I agree. There should be format validation for optional fields with any kind of value in it. If the value is incorrectly formatted, we should not allow the user to submit until it is reformatted or removed from the field, provided there is adequate helper text in the error message to help the user correct the format error.
